token/program-2022/src/pod.rs

```diff
@@ -59,6 +60,68 @@ impl From<OptionalNonZeroPubkey> for COption<Pubkey> {
     }
 }
 
+/// ElGamal public key used for encryption
+pub type EncryptionPubkey = pod::ElGamalPubkey;
+/// An EncryptionPubkey that encodes `None` as all `0`, meant to be usable as a Pod type.
+#[derive(Clone, Copy, Debug, Default, PartialEq, Pod, Zeroable)]
+#[repr(transparent)]
+pub struct OptionalNonZeroEncryptionPubkey(EncryptionPubkey);
+impl OptionalNonZeroEncryptionPubkey {
+    /// Checks equality between an OptionalNonZeroEncryptionPubkey and an EncryptionPubkey when
+    /// itnerpretted as bytes.
+    pub fn equals(&self, other: &EncryptionPubkey) -> bool {
+        &self.0 == other
+    }
+}
+impl TryFrom<Option<EncryptionPubkey>> for OptionalNonZeroEncryptionPubkey {
+    type Error = ProgramError;
+    fn try_from(p: Option<EncryptionPubkey>) -> Result<Self, Self::Error> {
+        match p {
+            None => Ok(Self(EncryptionPubkey::default())),
+            Some(encryption_pubkey) => {
+                if encryption_pubkey == EncryptionPubkey::default() {
+                    Err(ProgramError::InvalidArgument)
+                } else {
+                    Ok(Self(encryption_pubkey))
+                }
+            }
+        }
+    }
+}
+impl TryFrom<COption<EncryptionPubkey>> for OptionalNonZeroEncryptionPubkey {
+    type Error = ProgramError;
+    fn try_from(p: COption<EncryptionPubkey>) -> Result<Self, Self::Error> {
+        match p {
+            COption::None => Ok(Self(EncryptionPubkey::default())),
+            COption::Some(encryption_pubkey) => {
+                if encryption_pubkey == EncryptionPubkey::default() {
+                    Err(ProgramError::InvalidArgument)
+                } else {
+                    Ok(Self(encryption_pubkey))
+                }
+            }
+        }
+    }
+}
+impl From<OptionalNonZeroEncryptionPubkey> for Option<EncryptionPubkey> {
+    fn from(p: OptionalNonZeroEncryptionPubkey) -> Self {
+        if p.0 == EncryptionPubkey::default() {
+            None
+        } else {
+            Some(p.0)
+        }
+    }
+}
+impl From<OptionalNonZeroEncryptionPubkey> for COption<EncryptionPubkey> {
+    fn from(p: OptionalNonZeroEncryptionPubkey) -> Self {
+        if p.0 == EncryptionPubkey::default() {
+            COption::None
+        } else {
+            COption::Some(p.0)
+        }
+    }
+}
```

Comment:
And this? If it's not used, let's keep it out

Reply:
Yes, that is a good idea.